common/src/main/java/org/mvndaemon/mvnd/common/Environment.java

```diff
@@ -209,7 +209,7 @@
     /**
      * The <code>-Xmx</code> value to pass to the daemon
      */
-    MVND_MAX_HEAP_SIZE("mvnd.maxHeapSize", null, "2G", OptionType.MEMORY_SIZE, Flags.DISCRIMINATING),
+    MVND_MAX_HEAP_SIZE("mvnd.maxHeapSize", null, null, OptionType.MEMORY_SIZE, Flags.DISCRIMINATING | Flags.OPTIONAL),
```

Comment:
I assumed it's fine to keep the heap size as a discriminating flag still since the heap size cannot be changed on the fly and hence is useful to identify the client within the registry.